csrc/multidevice/pipeline.h

```diff
@@ -0,0 +1,142 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+#include <fusion.h>
+#include <ir/base_nodes.h>
+#include <multidevice/device_mesh.h>
+#include <multidevice/multidevice.h>
+#include <multidevice/pipeline_ir.h>
+
+/*
+This file implements the Pipeline interface.
+A Pipeline represents a Fusion segmented into a series of Stages, each Stage
```

Comment:
```suggestion
A Pipeline represents a Fusion or a parent stage segmented into a series of Stages, each Stage
```

Reply:
Ok, thanks for those suggestions. Anyway, once this PR is accepted we will, as you suggested, make the pipelining more general and compatible with task-parallelism, so we will reorganize and rewrite the comments by then